pkg/gktest/runner.go

```diff
@@ -17,25 +19,32 @@ type Runner struct {
 	// NewClient instantiates a Client for compiling Templates/Constraints, and
 	// validating objects against them.
 	NewClient func() (Client, error)
-
-	// TODO: Add Printer.
 }
 
 // Run executes all Tests in the Suite and returns the results.
-func (r *Runner) Run(ctx context.Context, filter Filter, s *Suite) SuiteResult {
+func (r *Runner) Run(ctx context.Context, filter Filter, suitePath string, s *Suite) SuiteResult {
+	start := time.Now()
 	result := SuiteResult{
+		Path:        suitePath,
 		TestResults: make([]TestResult, len(s.Tests)),
 	}
+	suiteDir := filepath.Dir(suitePath)
+
 	for i, t := range s.Tests {
 		if filter.MatchesTest(t) {
-			result.TestResults[i] = r.runTest(ctx, filter, t)
+			start := time.Now()
+			result.TestResults[i] = r.runTest(ctx, suiteDir, filter, t)
+			result.TestResults[i].Name = t.Name
+			result.TestResults[i].Runtime = Duration(time.Since(start))
```

Comment:
should the Name and Runtime fields be populated by `runTest`?

Reply:
There's too many places that runTest can exit. Since runTest isn't exported, it's okay to put that functionality in Run instead.